### DISCUSSION TITLE
porting to 32bit float

### DISCUSSION THREAD

**Discussion by leongzhang** [ivo-georgiev/ukfLib#14] [Ideas]
Hi, thanks for the work~
i see that the implementation here is with the double precision floats. I wonder if that was a choice based on accuracy or stability? have you tested/ported (or plan to test/port) to a 32bit float version? 
I'm trying to identify a possible path to implement ukf on an small ARM M series microcontroller with 32 bit floating point support but not much for the double precision. 
thanks!
gz

**Comment by ivo-georgiev** [comment]
Hi, you can try single precision behavior with simple change typedef on 39 line (double to become float) in System_Types.h.

**Reply by leongzhang** [reply]
> Hi, you can try single precision behavior with simple change typedef on 39 line (double to become float) in System_Types.h.

Hi, i just downloaded it and tried on the matlab (2016a) that i have. it seems with single precision it is broken. some states become NaN. 

(p.s. tried cfg1 and 2 both)

**Reply by ivo-georgiev** [reply]
Bad.... I have to check too.
